typed-racket-lib/typed-racket/typecheck/tc-expr-unit.rkt

```diff
@@ -387,22 +387,44 @@
      (match (and expected (resolve (intersect expected (-pair Univ Univ))))
        [(Pair: car-t cdr-t) (-pair (find-stx-type car car-t) (find-stx-type cdr cdr-t))]
        [_ (-pair (find-stx-type car) (find-stx-type cdr))])]
-    [(vector xs ...)
-     (match (and expected (resolve (intersect expected -VectorTop)))
-       [(Vector: t)
-        (make-Vector
-         (check-below
-          (apply Un
-                 (for/list ([x (in-list xs)])
-                   (find-stx-type x t)))
-          t))]
-       [(HeterogeneousVector: ts)
-        (make-HeterogeneousVector
-         (for/list ([x (in-list xs)]
-                    [t (in-list/rest ts #f)])
-           (cond-check-below (find-stx-type x t) t)))]
-       [_ (make-HeterogeneousVector (for/list ([x (in-list xs)])
-                                      (generalize (find-stx-type x #f))))])]
+    [(and (vector xs ...) v)
```

Comment:
Something strange is happening
- if I have phase 0 code that puts a _mutable_ vector into a syntax object, `syntax-e` returns an _immutable_ vector
- if I have phase 1 code that puts an _immutable_ vector into a syntax object, `syntax-e` returns a _mutable_  vector

Example:

```
#lang racket/base
(require rackunit (for-syntax rackunit racket/base))

(check-false ;; FAILS
 (immutable? (syntax-e #`#,(vector)))
 "(syntax-e (syntax (vector))) made immutable vector")

(check-true ;; PASSES
 (immutable? (syntax-e #`#,(vector-immutable)))
 "(syntax-e (syntax (vector-immutable))) made mutable vector")

(define-syntax (mvec stx)
  (define v (vector))
  (check-false ;; PASSES
   (immutable? v)
   "(vector) made immutable vector")
  #`#'#,v)

(check-false ;; PASSES
 (immutable? (syntax-e (mvec)))
 "(syntax-e (syntax (vector))) made immutable vector")

(define-syntax (ivec stx)
  (define v (vector-immutable))
  (check-true ;; PASSES
   (immutable? v)
   "(vector-immutable) made mutable vector")
  #`#'#,v)

(check-true ;; FAILS
 (immutable? (syntax-e (ivec)))
 "(syntax-e (syntax (vector-immutable))) made mutable vector")
```

Output:
```
--------------------
FAILURE
name:       check-false
location:   pr575-variation-3.rkt:4:0
params:     '(#t)
message:    "(syntax-e (syntax (vector))) made immutable vector"
--------------------
--------------------
FAILURE
name:       check-true
location:   pr575-variation-3.rkt:30:0
params:     '(#f)
message:    "(syntax-e (syntax (vector-immutable))) made mutable vector"
--------------------
```

I think this is a problem with Racket, so I've changed a few `typecheck-tests` to match the current (unexpected)  behavior to see if anything else goes wrong on Travis.